src/proof/theory_proof.cpp

```diff
@@ -869,6 +869,23 @@ void LFSCTheoryProofEngine::printBoundTerm(Expr term, std::ostream& os, const Pr
   printTheoryTerm(term, os, map);
 }
 
+void LFSCTheoryProofEngine::printBoundFormula(Expr term,
+                                              std::ostream& os,
+                                              const ProofLetMap& map)
+{
+  Assert(term.getType().isBoolean() or term.getType().isPredicate());
+  bool wrapWithBoolToPred = term.getType().isBoolean() and printsAsBool(term);
```

Comment:
https://github.com/CVC4/CVC4/blob/596fe8c79106dd9b7764df2ddce6b2d3344fea34/src/proof/theory_proof.cpp#L1113

Isn't `wrapWithBoolToPred` always false? What am I missing here?

Reply:
A term of sort `Bool` may, in the proof, be printed with type `formula`, in which case `printsAsBool` would return false, and `wrapWithBoolToPred` would also be false.